frameworks/non-keyed/kobold/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+    "short",
+    "long",
+    "handsome",
+    "plain",
+    "quaint",
+    "clean",
+    "elegant",
+    "easy",
+    "angry",
+    "crazy",
+    "helpful",
+    "mushy",
+    "odd",
+    "unsightly",
+    "adorable",
+    "important",
+    "inexpensive",
+    "cheap",
+    "expensive",
+    "fancy",
+];
+
+const COLOURS_LEN: usize = 11;
+static COLOURS: &[&str; COLOURS_LEN] = &[
+    "red", "yellow", "blue", "green", "pink", "brown", "purple", "brown", "white", "black",
+    "orange",
+];
+
+const NOUNS_LEN: usize = 13;
+const NOUNS: [&str; NOUNS_LEN] = [
+    "table", "chair", "house", "bbq", "desk", "car", "pony", "cookie", "sandwich", "burger",
+    "pizza", "mouse", "keyboard",
+];
+
+fn random(max: usize) -> usize {
+    (Math::random() * 1000.0) as usize % max
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct RowData {
+    id: usize,
+    label: String,
+}
+
+struct State {
+    rows: Vec<RowData>,
+    selected: Option<usize>,
+    last: usize,
+}
+
+impl State {
+    fn new() -> Self {
+        State {
+            rows: Vec::new(),
+            selected: None,
+            last: 0,
+        }
+    }
+    fn append_rows(&mut self, count: usize) {
+        self.rows.reserve_exact(count);
+        let d = self.last + 1;
+        for i in 0..count {
+            let adjective = ADJECTIVES[random(ADJECTIVES_LEN)];
+            let colour = COLOURS[random(COLOURS_LEN)];
+            let noun = NOUNS[random(NOUNS_LEN)];
+            let mut label = String::with_capacity(adjective.len() + colour.len() + noun.len() + 2);
+            label.push_str(adjective);
+            label.push(' ');
+            label.push_str(colour);
+            label.push(' ');
+            label.push_str(noun);
+            self.rows.push(RowData {
+                id: i + d,
+                label,
+            });
+        }
+        self.last += count;
+    }
+
+    fn remove(&mut self, id: usize) {
+        self.rows.remove(id);
+    }
+
+    fn run(&mut self) {
+        self.clear();
+        self.append_rows(1_000)
+    }
+    fn run_lots(&mut self) {
+        self.clear();
+        self.append_rows(10_000)
+    }
+    fn add(&mut self) {
+        self.append_rows(1_000)
+    }
+    fn update(&mut self) {
+        self.rows.iter_mut().step_by(10).for_each(|row| {
+            row.label.reserve_exact(4);
+            row.label.push_str(" !!!")
+        });
+    }
+    fn clear(&mut self) {
+        self.rows = Vec::new();
+        self.selected = None
+    }
+    fn swap(&mut self) {
+        if self.rows.len() > 998 {
+            self.rows.swap(1, 998);
+        }
+    }
+    fn select(&mut self, id: usize) {
+        self.selected = Some(id)
+    }
+}
+
+#[component]
+fn Row(num: usize, row: RowData, state: &Hook<State>) -> impl View {
+    let is_in_danger = class!("danger" if state.selected == Some(row.id));
+
+    bind! { state:
+        let remove = move |_| state.remove(num);
+        let select = move |_| state.select(row.id);
+    }
```

Comment:
I get an error

> borrowed data escapes outside of associated function
argument requires that `'a` must outlive `'static`

when making [these changes](https://github.com/krausest/js-framework-benchmark/pull/1239/files/163ee40ead8c03b3dc9799883944b1f982baf335#r1171233986)

Reply:
Right, the closure is trying to capture the entire row, copy the id on stack first:

```suggestion
    let id = row.id;
    
    bind! { state:
        let remove = move |_| state.remove(num);
        let select = move |_| state.select(id);
    }
```

Reply:
I fixed by adding let id = row.id.to_owned(); and using that instead of row.id;